src/event.rs

```diff
@@ -301,12 +301,14 @@ where
 		Self { queue, waker, kv_store, logger }
 	}
 
-	pub(crate) fn add_event(&self, event: Event) -> Result<(), Error> {
-		{
+	pub(crate) async fn add_event(&self, event: Event) -> Result<(), Error> {
```

Comment:
I was looking at what's left in EventQueue that is sync. Just `next_event`. Unrelated to this PR, but in every test where that method is used, the test is already async. So it could use the async version?

Reply:
Yes that would be an improvement indeed.